MyArrayList.java

```diff
@@ -1 +1,110 @@
+package com.company;
 
+public class MyArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int position = 0;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int size = array.length;
+
+    @Override
+    public void add(T value) {
+        checkForAdd(position);
+        add(value, position);
+    }
+
+    private void checkForAdd(int index) {
+        if (index < 0) {
+            throw new NoSuchFieldError();
+        }
+        if (index >= size) {
+            size += size / 2;
+            array = newArray();
+        }
+    }
+
+    private T[] newArray() {
+        Object[] newArray = new Object[size];
+        System.arraycopy(array, 0, newArray, 0, position);
+        return (T[]) newArray;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        checkForAdd(index);
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        array[index] = value;
+        position++;
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        Object[] obj = list.toArray();
+        array = newArray();
+        System.arraycopy(obj, 0, array, 0, obj.length);
```

Comment:
what does this line do?

Reply:
It copies the elements from obj into the array ('cause we want to get all the elements of the array obj and rewrite them into the current array)

Reply:
But I think i need to change the logic this method

Reply:
I did it without arraycopy

Reply:
please, view this implementation of the method
`if (array.length < sizeOfArray + list.size()) {
            Object[] newArr = (T[]) new Object[array.length + list.size()];
            System.arraycopy(array, 0, newArr, 0, array.length);
            System.arraycopy(list.toArray(), 0, newArr, array.length, list.size());
            array = newArr;
            sizeOfArray = sizeOfArray + list.size();
        } else {
            for (T value: list) {
                add(value);
            }
        }`